dfm/cleaning/quality.py

```diff
@@ -0,0 +1,199 @@
+"""
+Danish implementation of quality filter described in the paper:
+"Scaling Language Models: Methods, Analysis & Insights from Training Gopher"
+link: https://arxiv.org/abs/2112.11446
+
+Author:
+    Kenneth C. Enevoldsen
+    Kasper Junge
+
+References:
+    [1] Rae, J. W., Borgeaud, S., Cai, T., Millican, K., Hoffmann, J., Song, F., Aslanides, J., Henderson, S., Ring, R., Young, S., Rutherford, E.,
+    Hennigan, T., Menick, J., Cassirer, A., Powell, R., Driessche, G. van den, Hendricks, L. A., Rauh, M., Huang, P.-S., … Irving, G. (2021).
+    Scaling Language Models: Methods, Analysis & Insights from Training Gopher. https://arxiv.org/abs/2112.11446v2
+"""
+
+from collections import Counter
+from email.generator import Generator
+from functools import partial
+from typing import Iterable, Optional, Set, Tuple
+
+import spacy
+from spacy.tokens import Doc
+
+
+class QualityFilter:
+    def __init__(
+        self,
+        stop_words: Optional[Set[str]] = None,
+        min_stop_words: int = 2,
+        mean_word_length: Tuple[int, int] = (3, 10),
+        doc_length: Tuple[int, int] = [50, 100_000],
+        alpha_ratio: float = 0.7,  # changed from 70%
+        symbol_2_word_hashtag: float = 0.1,
+        symbol_2_word_ellipsis: float = 0.1,
+        max_p_begin_bullets: float = 0.9,
+        max_p_end_ellipsis: float = 0.3,
+    ):
+        """
```

Comment:
Normally the class docstring is included under the `class` line, rather than the `__init__` method, so I'd recommend moving yours.

Reply:
A yes, this is simply my tendency to follow the spaCy documentation, will move it